src/PlottingInterface_TEST.cc

```diff
@@ -0,0 +1,158 @@
+ * limitations under the License.
+ *
+*/
+#include <gtest/gtest.h>
+#include <ignition/msgs.hh>
+#include <ignition/transport.hh>
+#include <ignition/common/Console.hh>
+#include "test_config.h"
+#include "ignition/gui/Enums.hh"
+#include "ignition/gui/PlottingInterface.hh"
+
+using namespace ignition;
+using namespace gui;
+
+TEST(PlottingInterfaceTest, Topic)
+{
+    common::Console::SetVerbosity(4);
+
+    // ============== Register & UnRegister Test =============
+
+    // prepare the msg
+    msgs::Collision msg;
+    auto pose = new ignition::msgs::Pose();
+    auto vector3d = new ignition::msgs::Vector3d();
+    vector3d->set_x(10);
+    vector3d->set_z(15);
+    pose->set_allocated_position(vector3d);
+    msg.set_allocated_pose(pose);
+
+    auto topic = Topic("");
+
+    topic.Register("pose-position-x", 1);
+    topic.Register("pose-position-x", 2);
+    topic.Register("pose-position-y", 1);
+    topic.Register("pose-position-y", 2);
+    topic.UnRegister("pose-position-y", 2);
+
+    auto fields = topic.Fields();
+    ASSERT_EQ(topic.FieldCount(), 2);
+
+    // size test
+    EXPECT_EQ(fields["pose-position-x"]->ChartCount(), 2);
+    EXPECT_EQ(fields["pose-position-y"]->ChartCount(), 1);
+
+    // charts test
+    EXPECT_TRUE(fields["pose-position-x"]->Charts().find(1) !=
+            fields["pose-position-x"]->Charts().end());
+    EXPECT_TRUE(fields["pose-position-x"]->Charts().find(2) !=
+            fields["pose-position-x"]->Charts().end());
+    EXPECT_TRUE(fields["pose-position-y"]->Charts().find(1) !=
+            fields["pose-position-x"]->Charts().end());
+
+    // test the removing of the field if it has not attatched charts
+    topic.UnRegister("pose-position-y", 1);
+    EXPECT_EQ(topic.FieldCount(), 1);
+
+
+    // =========== Callback Test ============
+    topic.Register("pose-position-z", 1);
+
+    // update the fields
+    topic.Callback(msg);
+
+    fields = topic.Fields();
+
+    EXPECT_EQ(static_cast<int>(fields["pose-position-x"]->Value()), 10);
+    EXPECT_EQ(static_cast<int>(fields["pose-position-z"]->Value()), 15);
+}
+
+TEST(PlottingInterfaceTest, Transport)
+{
+    // =========== Publish Test =================
+    transport::Node node;
+
+    auto pub = node.Advertise<msgs::Collision> ("/collision_topic");
+    std::this_thread::sleep_for(std::chrono::milliseconds(200));
+
+    auto transport = Transport();
+    transport.Subscribe("/collision_topic", "pose-position-x", 1);
+    transport.Subscribe("/collision_topic", "pose-position-z", 1);
+
+    // prepare the msg
+    msgs::Collision msg;
+    auto pose = new ignition::msgs::Pose();
+    auto vector3d = new ignition::msgs::Vector3d();
+    vector3d->set_x(10);
+    vector3d->set_z(15);
+    pose->set_allocated_position(vector3d);
+    msg.set_allocated_pose(pose);
+
+
+    bool received = false;
+    std::function<void(const msgs::Collision &)> cb =
+        [&](const msgs::Collision &_msg)
+    {
+      EXPECT_EQ(_msg.pose().position().x(), 10);
```

Comment:
got a compiler warning here on Ubuntu CI
```
/home/claire/gsoc_dome_ws/src/ign-gui/test/gtest/include/gtest/gtest.h:1444:11: warning: comparing floating point with == or != is unsafe [-Wfloat-equal]
   if (lhs == rhs) {
```
I recommend to either cast the `x` value to `int`, or use `EXPECT_NEAR(_msg.pose().position().x(), 10, 1e-6)` to get rid of the warning